### DISCUSSION TITLE
Improving IPFS download speeds

### DISCUSSION THREAD

**Discussion by ltcompounder** [filecoin-project/community#524] [Q&A]
Hi everyone, I have used IPFS + pinata for a few projects and have run into an issue with serving files fast.

I would just like to understand a little bit more about IPFS and how download speeds can be improved.

Is this a matter of getting more people to start providing storage or does this have to do with something else?

**Comment by akhileshthite** [comment]
> Hi everyone, I have used IPFS + pinata for a few projects and have run into an issue with serving files fast.
> 
> I would just like to understand a little bit more about IPFS and how download speeds can be improved.
> 
> Is this a matter of getting more people to start providing storage or does this have to do with something else?

Hey @shon3005,
Use [nft.storage](https://nft.storage/) or [web3.storage](https://web3.storage/) according to your needs, both projects are part of Protocol Labs.
Both offer free services up to 1 TiB (after that you can request for more storage) and 31 GiB per upload. Download-upload speeds are fast and your content is permanently pinned with the help of Filecoin infrastructure.
